### ISSUE TITLE
Strange or missing notable ascents in profile

### ISSUE THREAD

**Issue by johnpitcairn** [theCrag/website#3292]
My profile used to show my best onsight (sport/trad 20) in a box at the top. This seems to have gone missing.

In notable ascents by style: 
I'm not seeing trad onsight or flash at all.
I never log a trad "tick" (which I am seeing here) I use "lead". Tick could mean anything.
Why am I seeing "top rope with rest" trad ascents? That sure isn't notable.

In notable ascents by area:
One of those Mt Eden Quarry ascents and two of the Araps ascents are top ropes - should those be there? The other Arapiles ascent is a second. My best Arapiles lead (onsight) is 19.

This seems to be a pretty misleading representation.

Example url to reproduce the problem:
https://www.thecrag.com/climber/johnpitcairn

Screenshot:
https://user-images.githubusercontent.com/6200850/47188079-0d348880-d393-11e8-8cdc-eb1869d14fb6.png

**Comment by johnpitcairn** [comment]
Aha, sorry I thought the release had been made, thanks.
Much better on my dev profile. The data does look to be quite old, will wait and see.

Remaining niggles:

- That "tick" when I never ever log anything as a simple tick. First an onsight, flash, redpoint, etc, then every clean lead after the first I log as "lead".
- It would be quite useful info to give each of those "by area" ascents a style and/or ascent type icon.
- Ideally it would be possible to filter those on the fly to exclude top ropes or seconds etc. I guess that's a separate issue.


![Dev profile](https://user-images.githubusercontent.com/6200850/47189847-fa25b680-d39a-11e8-9404-4ee9772bc7b3.png)

**Comment by scd** [comment]
Hi John,

We are aggregating for reporting purposes. The lead has been aggregated to a plain tick. Not sure what to do about this right now as there are some deep implications on what lead tick means (we would prefer people to log onsight, etc). Maybe we should open up a separate discussion on this altogether. Ultimately we may get rid of the lead tick style.

I don't plan to do much work in the area notable ticks because we plan to replace this with a map. This is pretty much just a place holder for future work. I will fix minor issues though.

We have had strong feedback that people want the best successful ascent recorded in notable ascents regardless of it's impressiveness.  We have got a strong preference to standardise the profile page with minimum profile configuration, which would potentially fix this issue for you. So we need to monitor this for a while before we make an assessment on which way to jump.

It looks like there is nothing specific for me to do right now. My suggestion is that you wait until the next release (planned 10 days away). After the release we can discuss what you think is important and create separate discussion issues.

Did you see the discussion about the profile page on the last release. It ended up being pretty constructive and useful.

**Comment by johnpitcairn** [comment]
> We have had strong feedback that people want the best successful ascent recorded in notable ascents regardless of it's impressiveness.

By area? Sure. An ascent type icon next to each of those would disambiguate that nicely.

> The lead has been aggregated to a plain tick. Not sure what to do about this right now as there are some deep implications on what lead tick means (we would prefer people to log onsight, etc). Maybe we should open up a separate discussion on this altogether. Ultimately we may get rid of the lead tick style.

Might be good. I'm curious whether you consider repeat clean leads to be a redpoint? I don't, only the first, as per the origin of the term. That doesn't leave anything other than "lead" or "tick" for subsequent clean leads. There are so many lazy default "tick" ascents in use that I want to differentiate clean repeat leads from those.

The increasing abuse of "all free" is another thing that warrants discussion - availability of that should really be region-specific.

**Comment by johnpitcairn** [comment]
A quick play with the html in Chrome, adding tick-type spans for notable ascents, produces this:

![Notable ascent ascent types](https://user-images.githubusercontent.com/6200850/47247065-c5c6fe80-d45d-11e8-9e0e-ded820508ea3.png)

**Comment by johnpitcairn** [comment]
Nice, cheers

**Comment by brendanheywood** [comment]
@johnpitcairn adding the tick type here has highlighted that these ticks may not be clean, I'm wondering if you thought that the hardest ticks under each area should be either filtered in some way, or perhaps sorted by grade and then tick type, or something else?

**Comment by johnpitcairn** [comment]
That "lead" on mine you mean? It's logged like that because I can't remember when I first climbed it, but in my case is clean. I guess for others it may not be though?

Personally I only want to see hardest clean ascents on my profile (and preferably leads), but others may be more interested in seeing the hardest thing they topped out on, however poorly (especially new/gym climbers). 

In my case I consider the onsighted Arapiles 19s much more relevant than the top rope flashed 21s. 

I don't think it should be automatically filtered any further, with no control over that. Sorting by tick type (onsight top) might be worthwhile for some, but that still won't help me for those Arapiles ascents if it is just looking at grade.

I also tend to look at others' profiles to assess whether I think I might get up a particular route, save it for an onsight attempt, etc. For that purpose I'd be most interested in being able to manually apply filters to others' profiles. The old "use my profile filters" option went some way towards that.

I'd see this as something like a "minimum style" filter(s) triggering an ajax reload of some or all of the notable ascents section. 

The problem I see is that conflating the the tick type (any, clean, flash, onsight etc) with the ascent style (any, solo, boulder, top rope, second, lead, aid, etc) produces a lot of possible combinations and edge cases. Addressing that globally might help deal with a number of other problems - I think there is another issue about that?

**Comment by scd** [comment]
Let's see how this goes. I have already made an update that only successful tick types will be shown in the crags notabable ascents.

One thing that I think is lost is that this is a public profile so having standards rather then configuration is very important. In other words it should be what you expect to see on other peoples profile. If we do configuration then it is likely that the settings will come from the account that is viewing your profile.

We do need to put more on your dashboard so you can see yourself.

**Comment by brendanheywood** [comment]
> conflating the the tick type with the ascent style

What do you mean by this exactly? The 'other issue' might be the analysis I did here:

https://github.com/theCrag/website/issues/1217#issuecomment-121139883

**Comment by johnpitcairn** [comment]
So this is still weird I reckon - I'm not seeing sport onsights (20) reflected in the notable ascents. Is that because the system has decided I'm a trad climber or is there a date cutoff or what?

![screen shot 2018-10-30 at 2 55 23 pm](https://user-images.githubusercontent.com/6200850/47690781-4ce15700-dc54-11e8-8716-a1eb80174552.png)

**Comment by scd** [comment]
Would you class yourself as a trad climber or sport climber?

**Comment by johnpitcairn** [comment]
Well either for the purposes of notable ascents. I'd expect grade 20 sport onsights to show up in preference to flashed trad 18s for sure. And I'd expect the onsighted sport 20s to be my "onsights" grade. It shouldn't get weighted in favour of trad just because 80% of my ascents are trad, if that's what it is doing?

**Comment by scd** [comment]
At the moment there is a lot of functionality tied to type of climber. 

Why should it not get weighted to your main style of climbing?

If you were at a party would you describe yourself as a sport climber or trad climber? If you said you were a trad climber would you then talk about your best sport routes?

**Comment by johnpitcairn** [comment]
If I got asked what I could onsight I would say 20. I think the system is trying to be too smart.

**Comment by scd** [comment]
If you went out to boulder one day and did a couple of boulder routes would you want it to say you onsight a V4?

It is the case where the system is not smart enough. It knows that it needs to separate the styles but now it must also know that this is only sometimes. 

I do get that you would like to say you onsight grade 20. We need to observer this for longer and see what other related issues there are.

**Comment by johnpitcairn** [comment]
Looks like I'd downgraded a soft trad 20 onsight to 18 at the time rather than accepting the community grade. 

But I'd at least still expect to see onsights listed in the "sport" section of those stats.

**Comment by johnpitcairn** [comment]
Totally my opinion of course ;-)

**Comment by scd** [comment]
>  No, but personally I consider bouldering as fundamentally pretty different to roped climbing, and the fact that the grading is different reflects that.

Yes this is what we may have missed in the last round of analysis. It is related to also wanting your trad and sport routes combined for your grade profile. You got to realise that the system thinks they are fundamentally different, and now we have got to tell the system that there are similarities in some situations.

Anyway we have to sit on this for a while so it can digest.

**Comment by lordyavin** [comment]
Don't know if this is the right issue, but looks like I got a bug with the style attribution of notable ascents by area:

![grafik](https://user-images.githubusercontent.com/1528540/56879642-ef3e9380-6a59-11e9-83d9-1a4ddada7b3b.png)

Problem here are the ['Via Andreas'](https://www.thecrag.com/climbing/germany/falkenwaende/route/249433707/ascents/by/captainhook) and the ['Bierkante'](https://www.thecrag.com/climbing/germany/falkenwaende/route/249433707/ascents/by/captainhook). The hardest ascents are displayed as Top rope but actually I red pointed both in the meantime but with my red point ascent I downgraded the route and didn't update all previous attempts. Therefore the TR ascents win over the RP.
